messaginginapp/src/main/java/io/customer/messaginginapp/gist/presentation/GistModalActivity.kt

```diff
@@ -77,32 +85,49 @@ class GistModalActivity : AppCompatActivity(), GistListener, GistViewListener, T
                     MessagePosition.TOP -> binding.modalGistViewLayout.setVerticalGravity(Gravity.TOP)
                 }
             }
-        } ?: run {
-            finish()
         }
 
+        subscribeToAttributes()
+
         // Update back button to handle in-app message behavior, disable back press for persistent messages, true otherwise
-        val onBackPressedCallback = object : OnBackPressedCallback(isPersistentMessage()) {
+        val onBackPressedCallback = object : OnBackPressedCallback(isPersistentMessage(parsedMessage)) {
             override fun handleOnBackPressed() {}
         }
         onBackPressedDispatcher.addCallback(this, onBackPressedCallback)
     }
 
-    override fun onResume() {
-        inAppMessagingManager.dispatch(InAppMessagingAction.LogEvent("GisModelActivity onResume"))
-        super.onResume()
-        GistSdk.addListener(this)
+    private fun subscribeToAttributes() {
+        attributesListenerJob.add(
+            inAppMessagingManager.subscribeToAttribute(
+                selector = { it.currentMessageState },
+                areEquivalent = { old, new ->
+                    when {
+                        old is MessageState.Default && new is MessageState.Default -> true
+                        old is MessageState.Loaded && new is MessageState.Loaded -> old.message == new.message
+                        old is MessageState.Dismissed && new is MessageState.Dismissed -> old.message == new.message
+                        old is MessageState.Processing && new is MessageState.Processing -> old.message == new.message
+                        else -> false
+                    }
+                }
+            ) { state ->
+                if (state is MessageState.Loaded) {
+                    onMessageShown(state.message)
+                }
+
+                if (state is MessageState.Dismissed) {
+                    cleanUp()
+                }
+            }
+        )
     }
 
     override fun onPause() {
-        inAppMessagingManager.dispatch(InAppMessagingAction.LogEvent("GisModelActivity onPause"))
-        GistSdk.removeListener(this)
         super.onPause()
         overridePendingTransition(0, 0)
     }
 
     override fun finish() {
-        inAppMessagingManager.dispatch(InAppMessagingAction.LogEvent("GisModelActivity finish"))
+        logger.debug("GisModelActivity finish")
```

Comment:
Not for this PR though, but noticed typo in `GisModelActivity` log in whole. Something that we can note and update later.